### PR TITLE
reef: crimson/os/seastore: evict cold data to slower devices

### PR DESCRIPTION
**Pull request by Matan-B**
This PR is part of Crimson Reef backport batch, See: https://gist.github.com/Matan-B/0e076b8c55545c631012bb22a996b6e6

---

backport of https://github.com/ceph/ceph/pull/47974

this backport was staged using crimson-backport.sh which is based on ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

File: doc/dev/crimson/crimson.rst [hunk 1]
```diff
@@ -154,16 +154,16 @@ To facilitate the development of crimson, following options would be handy when
 using ``vstart.sh``,
 
 ``--crimson``
-    start ``crimson-osd`` instead of ``ceph-osd``
+    Start ``crimson-osd`` instead of ``ceph-osd``.
 
 ``--nodaemon``
-    do not daemonize the service
+    Do not daemonize the service.
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:160]: or coal?

File: doc/dev/crimson/crimson.rst [hunk 2]
```diff
@@ -154,16 +154,16 @@ To facilitate the development of crimson, following options would be handy when
...
 
 ``--redirect-output``
-    redirect the stdout and stderr of service to ``out/$type.$num.stdout``.
+    Tedirect the stdout and stderr of service to ``out/$type.$num.stdout``.
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:163]: Yes I know this is a backport. Note to self:  fix.

File: doc/dev/crimson/crimson.rst [hunk 3]
```diff
@@ -154,16 +154,16 @@ To facilitate the development of crimson, following options would be handy when
...
 
 ``--osd-args``
-    pass extra command line options to crimson-osd or ceph-osd. It's quite
+    Pass extra command line options to crimson-osd or ceph-osd. It's quite
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:166]: NTS: backticks

File: doc/dev/crimson/crimson.rst [hunk 4]
```diff
@@ -173,14 +173,31 @@ using ``vstart.sh``,
     for more Seastar specific command line options.
 
 ``--cyanstore``
-    use CyanStore as the object store backend.
+    Use CyanStore as the object store backend.
 
 ``--bluestore``
-    use the alienized BlueStore as the object store backend. This is the default
+    Use the alienized BlueStore as the object store backend. This is the default
```
Comment (anthonyeleven) [inline doc/dev/crimson/crimson.rst:179]: NTS: period

File: src/common/options/crimson.yaml.in [hunk 5]
```diff
@@ -102,3 +102,18 @@ options:
   level: dev
   desc: Total size to use for CircularBoundedJournal if created, it is valid only if seastore_main_device_type is RANDOM_BLOCK
   default: 5_G
+- name: seastore_multiple_tiers_stop_evict_ratio
+  type: float
+  level: advanced
+  desc: When the used ratio of main tier is less than this value, then stop evict cold data to the cold tier.
```
Comment (anthonyeleven) [inline src/common/options/crimson.yaml.in:108]: NTS: the main tier

File: src/vstart.sh [hunk 6]
```diff
@@ -255,7 +256,8 @@ options:
 	--no-restart: dont restart process when using ceph-run
 	--jaeger: use jaegertracing for tracing
 	--seastore-devs: comma-separated list of blockdevs to use for seastore
-	--seastore-secondary-des: comma-separated list of secondary blockdevs to use for seastore
+	--seastore-secondary-devs: comma-separated list of secondary blockdevs to use for seastore
```
Comment (anthonyeleven) [inline src/vstart.sh:259]: nts: camelcase

**Review by anthonyeleven — APPROVED** [review]
Docs approved given backport

**Review by athanatos — APPROVED** [review]

**Comment by Matan-B** [comment]
Rebased, only a doc was conflicting due to: #51677. Merging.

**Pull request merged**